### PR TITLE
test: refactor header inclusion to speed up building (for test/common/...)

### PR DESCRIPTION
**Pull request by foreseeable**
Signed-off-by: Muge Chen <mugechen@google.com>

Refactor mock class include directive after dividing test/server/mocks.h (#11649)

Since the refactoring affects ~100 files so I split it up to several pull requests (previous one #11952 #12045)

Building time comparison for affected tests: see [spreadsheet](https://docs.google.com/spreadsheets/d/e/2PACX-1vTx4OuTC1H_KFaEOxLxeUlEU70s8N8W6VDC7P5Vj9nk3mvA0263gPE-1hwFPgYwfmusTsUHyUjdyoZe/pubhtml) at #10917 comment

Commit Message: refactor header inclusion to speed up building
Additional Description:
Risk Level: low
Testing: exsiting tests
Docs Changes: N/A
Release Notes: no
Related Issues: #10917

\cc @ahedberg @bbarenblat

### CODE REVIEW

File: test/common/grpc/grpc_client_integration_test_harness.h [hunk 1]
```diff
@@ -28,7 +29,8 @@
 #include "test/integration/fake_upstream.h"
 #include "test/mocks/grpc/mocks.h"
 #include "test/mocks/local_info/mocks.h"
-#include "test/mocks/server/mocks.h"
+#include "test/mocks/server/transport_socket_factory_context.h"
+
```
Comment (ahedberg) [inline test/common/grpc/grpc_client_integration_test_harness.h:33]: minor: remove the extra newline here? (I'd expect the fix-formatting script to take care of this.)

Comment (foreseeable) [reply test/common/grpc/grpc_client_integration_test_harness.h]: fix-formatting didn't remove newline like this. It has been removed manually.

**Comment by junr03** [comment]
@foreseeable the failure here seems potentially real.

File: test/common/grpc/grpc_client_integration_test_harness.h [hunk 2]
```diff
@@ -28,7 +29,7 @@
 #include "test/integration/fake_upstream.h"
 #include "test/mocks/grpc/mocks.h"
 #include "test/mocks/local_info/mocks.h"
-#include "test/mocks/server/mocks.h"
+#include "test/mocks/server/transport_socket_factory_context.h"
```
Comment (ahedberg) [inline test/common/grpc/grpc_client_integration_test_harness.h:32]: `Grpc::StatNames` is an incomplete type here (https://dev.azure.com/cncf/envoy/_build/results?buildId=44924&view=logs&j=ec9f720e-f904-5f65-4ba4-f371a013344f&t=ee47c280-92b1-5c80-3a10-d3d23a58a062&l=3606); we need to #include the header providing that name. server/mocks.h probably indirectly #included it before.

Comment (foreseeable) [reply test/common/grpc/grpc_client_integration_test_harness.h:33]: yes, server/mocks.h previously #include "common/grpc/context_impl.h" which providing that name. i have sent a pr resolve this.

**Pull request merged**